tastypie/resources.py

```diff
@@ -2233,12 +2233,12 @@ def resolve_filter_type(self, field_name, filter_bits, default_filter_type=None)
         if not filter_bits:
             # No filter type to resolve, use default
             return default_filter_type
-        elif filter_bits[0] not in self.get_query_terms(field_name):
+        elif filter_bits[-1] not in self.get_query_terms(field_name):
```

Comment:
For nested objects, the first element is the field on which the filter needs to be applied. Instead, a filter type should be used.

Reply:
@swarupbam you are right, but the problem is a bit deeper than with just using `[-1]` here. The current field might still not have the required filter in `self.get_query_terms`. Instead, we need to recursively traverse the chain of related objects (in the `filter_bits`) and properly ask the latest one in the chain. See the fix implementation here: https://github.com/Partender/django-tastypie/pull/1